domain/src/main/java/com/sanket/domain/MeasureHeartRateService.kt

```diff
@@ -0,0 +1,133 @@
+            stopSelf()
+        }
+        Log.e("SERVICE::",":::STARTED")
+        generateForegroundNotification()
+        startListeningHeartRate()
+        return START_STICKY
+    }
+
+    @SuppressLint("NewApi")
+    private fun startListeningHeartRate() {
+        sensorManager = (getSystemService(SENSOR_SERVICE) as SensorManager)
+        heartRateSensonr = sensorManager.getDefaultSensor(Sensor.TYPE_HEART_RATE)
+        sensorManager.registerListener(this, heartRateSensonr, SensorManager.SENSOR_DELAY_FASTEST)
+    }
+
+    private fun generateForegroundNotification() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val pendingIntent =
+                PendingIntent.getActivity(this, 0, Intent(), 0)
+            iconNotification = BitmapFactory.decodeResource(resources, R.mipmap.ic_launcher)
+            if (mNotificationManager == null) {
+                mNotificationManager =
+                    this.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            }
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                assert(mNotificationManager != null)
+                mNotificationManager?.createNotificationChannelGroup(
+                    NotificationChannelGroup("heart_rate_measure", "HeartRateMeasure")
+                )
+                val notificationChannel =
+                    NotificationChannel(
+                        "service_channel", "Service Notifications",
+                        NotificationManager.IMPORTANCE_MIN
+                    )
+                notificationChannel.enableLights(false)
+                notificationChannel.lockscreenVisibility = Notification.VISIBILITY_SECRET
+                mNotificationManager?.createNotificationChannel(notificationChannel)
+            }
+            val builder = NotificationCompat.Builder(this, "service_channel")
+
+            builder.setContentTitle(
+                StringBuilder(resources.getString(R.string.app_name)).append(" service is running")
+                    .toString()
+            )
+                .setTicker(
+                    StringBuilder(resources.getString(R.string.app_name)).append("service is running")
+                        .toString()
+                )
+                .setContentText("Measuring heart rate")
+                .setSmallIcon(R.mipmap.ic_launcher)
+                .setPriority(NotificationCompat.PRIORITY_LOW)
+                .setWhen(0)
+                .setOnlyAlertOnce(true)
+                .setContentIntent(pendingIntent)
+                .setOngoing(true)
+            if (iconNotification != null) {
+                builder.setLargeIcon(Bitmap.createScaledBitmap(iconNotification!!, 128, 128, false))
+            }
+            builder.color = resources.getColor(R.color.menu_item_text_color_selected, null)
+            notification = builder.build()
+            startForeground(mNotificationId, notification)
```

Comment:
please add OngoingActivity here

Reply:
@EntyR right now this foreground service is placed at data layer, and data layer does not have access of presentation module so On going activity can not be added here. So pending intent will launch the default launcher activity of the application when notification is clicked by user.